sycl/test/basic_tests/diagnostics/non-uniform-wk-gp-test.cpp

```diff
@@ -0,0 +1,70 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+// XFAIL: opencl
+// REQUIRES: level_zero
+//==------------------- non-uniform-wk-gp-test.cpp
+//--------------------------==//
+// This is a diagnostic test which verifies that for loops with non-uniform work
+//  groups size errors are handled correctly.
+//==--------------------------------------------------------------------------==//
+
+#include <CL/sycl.hpp>
+#include <iostream>
+
+using namespace cl::sycl;
+
+int test() {
+  try {
+    queue q = queue();
+    auto device = q.get_device();
+    auto deviceName = device.get_info<cl::sycl::info::device::name>();
+    std::cout << " Device Name: " << deviceName << std::endl;
+
+    const int N = 1;
+    {
+      q.submit([&](handler &cgh) {
+        cgh.parallel_for<class test_kernel>(
+            nd_range<3>(range<3>{1, 1, N}, range<3>{1, 1, 16}),
+            [=](nd_item<3> itm) {
+              int i = itm.get_global_id(0);
+              int j = itm.get_global_id(1);
+              int k = itm.get_global_id(2);
```

Comment:
`i`, `j`, and `k` are not used. Doesn't compiler yield a warning here?

Reply:
Replaced with a new code.